src/cdk/testing/testbed/task-state-zone-interceptor.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {BehaviorSubject, Observable} from 'rxjs';
+import {ProxyZone, ProxyZoneStatic, HasTaskState, Zone, ZoneDelegate} from './zone-types';
+
+/** Current state of the intercepted zone. */
+export interface TaskState {
+  /** Whether the zone is stable (i.e. no microtasks and macrotasks). */
+  stable: boolean;
+}
+
+/** Unique symbol that is used to patch a property to a proxy zone. */
+const stateObservableSymbol = Symbol('ProxyZone_PATCHED#stateObservable');
+
+/** Type that describes a potentially patched proxy zone instance. */
+type PatchedProxyZone = ProxyZone & {
+  [stateObservableSymbol]: undefined|Observable<TaskState>;
+};
+
+/**
```

Comment:
I would just leave a TODO here explaining that this is a temporary workaround for the fact that there is no way to wait for tasks outside of NgZone and link to https://github.com/angular/angular/issues/32896